lib/services/local/presence.go

```diff
@@ -752,44 +818,87 @@ func (s *PresenceService) GetRemoteClusters(opts ...services.MarshalOption) ([]t
 	return clusters, nil
 }
 
-// getRemoteCluster returns a remote cluster in raw form and unmarshaled
-func (s *PresenceService) getRemoteCluster(ctx context.Context, clusterName string) (*backend.Item, types.RemoteCluster, error) {
+// ListRemoteClusters returns a page of remote clusters
+func (s *PresenceService) ListRemoteClusters(
+	ctx context.Context, pageSize int, pageToken string,
+) ([]types.RemoteCluster, string, error) {
+	rangeStart := backend.Key(remoteClustersPrefix, pageToken)
+	rangeEnd := backend.RangeEnd(backend.ExactKey(remoteClustersPrefix))
+
+	// Adjust page size, so it can't be too large.
+	if pageSize <= 0 || pageSize > apidefaults.DefaultChunkSize {
+		pageSize = apidefaults.DefaultChunkSize
+	}
+
+	limit := pageSize + 1
+
+	result, err := s.GetRange(ctx, rangeStart, rangeEnd, limit)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	clusters := make([]types.RemoteCluster, 0, len(result.Items))
+	for _, item := range result.Items {
+		cluster, err := services.UnmarshalRemoteCluster(item.Value,
+			services.WithResourceID(item.ID),
+			services.WithExpires(item.Expires),
+			services.WithRevision(item.Revision),
+		)
+		if err != nil {
+			s.log.WithError(err).WithField("key", item.Key).Warn("Skipping item during ListRemoteClusters because conversion from backend item failed")
+			continue
+		}
+		clusters = append(clusters, cluster)
+	}
+
+	next := ""
+	if len(clusters) > pageSize {
+		next = backend.GetPaginationKey(clusters[pageSize])
+		// Truncate the last item that was used to determine next row existence.
+		clusters = clusters[:pageSize]
```

Comment:
Should we also add a `clear(clusters[pageSize:])`, just to be tidy?

Reply:
Erm - wouldn't that delete all of the entries in the slice we're about to return ? I've not had a chance to get familiar with `clear()` yet.

Reply:
No, like:
```suggestion
		clear(clusters[pageSize:])
		clusters = clusters[:pageSize]
```

Reply:
26409563e7